projects/storefrontlib/src/lib/product/components/product-list/product-view/product-view.component.scss

```diff
@@ -1,143 +1,97 @@
+  color: var(--cx-g-color-light);
 
-    @include media-breakpoint-down(sm) {
-      padding: $cx-product-search-layout-grid-mobile-trbl-padding;
-    }
+  @include media-breakpoint-down(sm) {
+    padding: var(--cx-padding, 13px 12px 13px 12px);
+  }
 
-    span {
-      @include var-color('color', $cx-product-search-layout-color);
-      @include var-color(
-        'background',
-        $cx-product-search-layout-background-color
-      );
+  span {
+    color: var(--cx-g-color-inverse);
+    background-color: var(--cx-g-color-secondary);
 
-      &:hover {
-        @include var-color(
-          'background',
-          $cx-product-search-layout-span-hover-background-color
-        );
-      }
+    &:hover {
+      background: var(--cx-g-color-primary);
+    }
 
-      &:before {
-        -webkit-transform: rotate(90deg);
-        -moz-transform: rotate(90deg);
-        -ms-transform: rotate(90deg);
-        -o-transform: rotate(90deg);
-        transform: rotate(90deg);
-        content: '';
-        position: absolute;
-        height: $cx-product-search-layout-grid-span-before-height;
-        width: $cx-product-search-layout-grid-span-before-width;
-        top: $cx-product-search-layout-grid-span-before-top;
-        left: $cx-product-search-layout-grid-span-before-left;
-        margin: $cx-product-search-layout-grid-span-before-trbl-margin;
-        @include var-color(
-          'background-color',
-          $cx-product-search-layout-grid-span-before-background-color
-        );
-      }
+    &:before {
+      -webkit-transform: rotate(90deg);
+      -moz-transform: rotate(90deg);
+      -ms-transform: rotate(90deg);
+      -o-transform: rotate(90deg);
+      transform: rotate(90deg);
+      content: '';
+      position: var(--cx-position, absolute);
+      width: var(--cx-width, 100%);
+      height: var(--cx-height, 2px);
+      top: var(--cx-top, 50%);
+      left: var(--cx-left, 0);
+      margin: var(--cx-margin, -1px 0 0 0);
+      background-color: var(--cx-g-color-inverse);
+    }
 
-      &:after {
-        -webkit-transform: rotate(0deg);
-        -moz-transform: rotate(0deg);
-        -ms-transform: rotate(0deg);
-        -o-transform: rotate(0deg);
-        transform: rotate(0deg);
-        content: '';
-        position: absolute;
-        height: $cx-product-search-layout-grid-span-before-height;
-        width: $cx-product-search-layout-grid-span-before-width;
-        top: $cx-product-search-layout-grid-span-before-top;
-        left: $cx-product-search-layout-grid-span-before-left;
-        margin: $cx-product-search-layout-grid-span-before-trbl-margin;
-        @include var-color(
-          'background-color',
-          $cx-product-search-layout-grid-span-before-background-color
-        );
-      }
+    &:after {
+      -webkit-transform: rotate(0deg);
+      -moz-transform: rotate(0deg);
+      -ms-transform: rotate(0deg);
+      -o-transform: rotate(0deg);
+      transform: rotate(0deg);
+      content: '';
```

Comment:
One idea I had is that since `:after` and `:before` share some styling (except the transform) perhaps we could do something like.

```
%beforeAfter {
     content: ' ';
      position: var(--cx-position, absolute);

      position: var(--cx-position, absolute);
      width: var(--cx-width, 100%);
      height: var(--cx-height, 2px);
      top: var(--cx-top, 50%);
      left: var(--cx-left, 0);
      margin: var(--cx-margin, -1px 0 0 0);
      background-color: var(--cx-g-color-inverse);
}
```

And then @extend the placeholder in the before and after. What do you think?

Reply:
What do you think?